File: Array/717. 1-bit and 2-bit Characters.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        i, past, n, = 0, 1, len(bits)
        while i < n:
            if bits[i]:
                past = 2
                i += 1
            else:
                past = 1
            i += 1
        return True if past == 1 else False

File: 717. 1-bit and 2-bit Characters.cpp
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) 
    {
        int isOneBit = 1, i = 0, n = bits.size();
        while(i < n) 
        {
            if(bits[i]) 
            {
                isOneBit = 0;
                i += 1;
            }
            else isOneBit = 1;
            i += 1;
        }   
        return isOneBit; 
    }
};